src/Microsoft.Framework.Primitives/StringValues.cs

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Primitives
+{
+    /// <summary>
+    /// Represents zero/null, one, or many strings in an efficient way.
+    /// </summary>
+    public struct StringValues : IList<string>
+    {
+        private static readonly string[] EmptyArray = new string[0];
+        public static readonly StringValues Empty = new StringValues(EmptyArray);
+
+        private readonly string _value;
+        private readonly string[] _values;
+
+        public StringValues(string value)
+        {
+            _value = value;
+            _values = null;
+        }
+
+        public StringValues(string[] values)
+        {
+            _value = null;
+            _values = values;
+        }
+
+        public static implicit operator StringValues(string value)
+        {
+            return new StringValues(value);
+        }
+
+        public static implicit operator StringValues(string[] values)
+        {
+            return new StringValues(values);
+        }
+
+        public static implicit operator string (StringValues values)
+        {
+            return values.GetStringValue();
+        }
+
+        public static implicit operator string[] (StringValues value)
```

Comment:
This returns the private array. Might be surprising to many people. Maybe it should be a method or the return type should be one of the red-only collection wrappers?